src/profile-logic/call-tree.js

```diff
@@ -362,6 +363,54 @@ export class CallTree {
       this._thread
     );
   }
+
+  /**
+   * Take a CallNodeIndex, and compute an inverted path for it.
+   *
+   * e.g:
+   *   (invertedPath, invertedCallTree) => path
+   *   (path, callTree) => invertedPath
+   *
+   * Call trees are sorted with the CallNodes with the heaviest total time as the first
+   * entry. This function walks to the tip of the heaviest branches to find the leaf node,
+   * then construct an inverted CallNodePath with the result. This gives a pretty decent
+   * result, but it doesn't guarantee that it will select the heaviest CallNodePath for the
+   * INVERTED call tree. This would require doing a round trip through the reducers or
+   * some other mechanism in order to first calculate the next inverted call tree. This is
+   * probably not worth it, so go ahead and use the uninverted call tree, as it's probably
+   * good enough.
+   */
+  findHeavyPathToSameFunctionAfterInversion(
+    callNodeIndex: IndexIntoCallNodeTable | null
+  ): CallNodePath {
+    if (callNodeIndex === null) {
+      return [];
+    }
+    let parentSelf = 0;
+    let children = [callNodeIndex];
+    const pathToLeaf = [];
+    do {
+      // Walk down the tree's depth to construct a path to the leaf node, this should
+      // be the heaviest branch of the tree.
+      const firstChild = children[0];
+      const { total, self } = this.getNodeData(firstChild);
+      if (total < parentSelf) {
+        break;
+      }
+      callNodeIndex = firstChild;
```

Comment:
Writing the comment made me realize that I was making an assumption which isn't necessarily true for diff profiles. I didn't fix that case, because I'm planning to rewrite this function soon anyway.

Here's the comment I'm adding:

```js
      if (total < parentSelf) {
        // The parent node was the node with the highest self time. We don't need
        // to descend any deeper because we won't find another node with higher
        // self time in this subtree because this subtree's total is already lower.
        // ... except if we have negative weights, i.e. a diff profile, but we
        // don't bother to handle those separately.
        break;
      }
```

Reply:
(The rewritten version will just check the self time of all the call nodes for this subtree, and pick the call node with the highest self time. To iterate just the call nodes of a subtree, it will take advantage of call nodes being ordered in tree order, so I'll need to land those "tree order" patches first.)